app/services/billable_metrics/aggregations/unique_count_service.rb

```diff
@@ -4,29 +4,41 @@ module BillableMetrics
   module Aggregations
     class UniqueCountService < BillableMetrics::Aggregations::BaseService
       def aggregate(from_datetime:, to_datetime:, options: {})
-        events = events_scope(from_datetime:, to_datetime:)
-          .where("#{sanitized_field_name} IS NOT NULL")
+        @from_datetime = from_datetime
+        @to_datetime = to_datetime
 
-        result.aggregation = events.count("DISTINCT (#{sanitized_field_name})")
-        result.pay_in_advance_aggregation = BigDecimal(compute_pay_in_advance_aggregation(events))
+        result.aggregation = compute_aggregation.ceil(5)
+        result.pay_in_advance_aggregation = BigDecimal(compute_pay_in_advance_aggregation)
         result.options = { running_total: running_total(options) }
-        result.count = events.count
+        result.count = result.aggregation
         result
       end
 
-      def compute_pay_in_advance_aggregation(events)
+      def compute_pay_in_advance_aggregation
         return 0 unless event
         return 0 if event.properties.blank?
 
-        existing_property = events
-          .where("#{sanitized_field_name} = '#{event.properties[billable_metric.field_name]}'")
-          .where.not(id: event.id)
-          .any?
+        unless previous_event
+          res = operation_type == :add ? 1 : 0
+          handle_event_metadata(current_aggregation: res, max_aggregation: res)
 
-        # NOTE: bill only property that have not been received in the period yet
-        return 0 if existing_property
+          return res
+        end
 
-        1
+        old_aggregation = BigDecimal(previous_event.metadata['current_aggregation'])
+        old_max = BigDecimal(previous_event.metadata['max_aggregation'])
+
+        current_aggregation = operation_type == :add ? (old_aggregation + 1) : (old_aggregation - 1)
+
+        if current_aggregation > old_max
+         handle_event_metadata(current_aggregation:, max_aggregation: current_aggregation)
+
+         1
+        else
+         handle_event_metadata(current_aggregation:, max_aggregation: old_max)
```

Comment:
Layout/IndentationWidth: Use 2 (not 1) spaces for indentation. (https://rubystyle.guide#spaces-indentation)